js/plugins/vertexai/src/imagen.ts

```diff
@@ -161,10 +187,19 @@ function toParameters(
   return out;
 }
 
-function extractPromptImage(request: GenerateRequest): string | undefined {
+function extractMaskImage(request: GenerateRequest): string | undefined {
+  return request.messages
+    .at(-1)
+    ?.content.find((p) => !!p.media && p.metadata?.type === 'mask')
+    ?.media?.url.split(',')[1];
+}
+
+function extractBaseImage(request: GenerateRequest): string | undefined {
   return request.messages
     .at(-1)
-    ?.content.find((p) => !!p.media)
+    ?.content.find(
+      (p) => !!p.media && (!p.metadata?.type || p.metadata?.type === 'base')
+    )
```

Comment:
Is this metadata to be set by the end-user? Will we follow up w/ a docs update?

Reply:
yes.  added docs.